src/libraries/Microsoft.Extensions.Hosting/src/HostBuilder.cs

```diff
@@ -349,6 +351,11 @@ private void InitializeServiceProvider()
                 configureServicesAction(_hostBuilderContext!, services);
             }
 
+            if (_hostBuilderContext!.HostingEnvironment.IsDevelopment() && _defaultProviderFactoryUsed)
```

Comment:
So the check for `IsDevelopment` is due to this: https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Hosting/src/HostingHostBuilderExtensions.cs#L327

Reply:
My understanding is that we don't want those values set to true by default except in development for performance reasons.